.github/workflows/publish.yml

```diff
@@ -1,57 +1,54 @@
 # This workflow is triggered by a new release on GitHub and then uploads
 # MulensModel package to PyPI.
-#
-# If the publish step finishes succesfully, the source distribution is
-# guaranteed to be published. Binary distributions are published only if
-# their respective build steps passes. If any binary distribution fails,
-# package maintainers may manually debug, build, and publish without the
-# automated workflow.
 name: Upload Python Package to PyPI
 
 on:
   release:
     types: [published]
+  workflow_dispatch:
 
 jobs:
-  build_binary_manylinux:
-    runs-on: ubuntu-latest
-    container: quay.io/pypa/manylinux_2_24_x86_64
-
+  build_wheels:
+    name: Build wheel [${{ matrix.os }} ${{ matrix.cw_archs }} ${{ matrix.cw_build }}]
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        python-version:
-          - cp37-cp37m
-          - cp38-cp38
-          - cp39-cp39
-          - cp310-cp310
+        # A matrix of OSs, Python verisons, and architectures
+        # Skips PyPy
+        os: [ubuntu-latest, macos-latest]
+        cw_build: [cp37-*, cp38-*, cp39-*, cp310-*, cp311-*]
+        include:
+        # Linux x86_64
+        - {os: ubuntu-latest, cw_archs: x86_64, cw_skip: "pp* *_i686"}         
+        # MacOS Intel
+        - {os: macos-latest, cw_archs: "x86_64", cw_skip: "pp*"}
+        # MacOS Silicon (>= Python 3.8 only)
+        - {os: macos-latest, cw_archs: "arm64", cw_build: "cp38-*, cp39-*, cp310-*, cp311-*", cw_skip: "pp*"}
```

Comment:
It looks like wheel metadata doesn't exists for MacOS Silicon before Python 3.8.